flask/src/main/java/net/corda/gradle/flask/FlaskJarTask.java

```diff
@@ -0,0 +1,235 @@
+            zoos.closeEntry();
+            for(FileCopyDetails fileCopyDetails : fileCopyDetailsInternals) {
+                String entryName = fileCopyDetails.getRelativePath().toString();
+                if(Objects.equals(Flask.Constants.METADATA_FOLDER, entryName)) continue;
+                zipEntry = new ZipEntry(entryName);
+                if(fileCopyDetails.isDirectory()) {
+                    zipEntry = new ZipEntry(entryName + '/');
+                    zoos.putNextEntry(zipEntry);
+                } else {
+                    boolean compressed = Flask.splitExtension(fileCopyDetails.getSourceName())
+                            .map(entry -> Objects.equals(".jar", entry.getValue()))
+                            .orElse(false);
+                    if (!compressed) {
+                        zipEntry.setMethod(ZipEntry.DEFLATED);
+                    } else {
+                        try (InputStream is = new FileInputStream(fileCopyDetails.getFile())) {
+                            Flask.computeSizeAndCrc32(zipEntry, is, buffer);
+                        }
+                        zipEntry.setMethod(ZipEntry.STORED);
+                    }
+                    try(InputStream is = new FileInputStream(fileCopyDetails.getFile())) {
+                        zoos.putNextEntry(zipEntry);
+                        Flask.write2Stream(zoos, is, buffer);
+                        zoos.closeEntry();
+                    }
+                }
+            }
+        }
+
+        @Override
+        @SneakyThrows
+        public void processFile(FileCopyDetailsInternal fileCopyDetailsInternal) {
+            String entryName = fileCopyDetailsInternal.getRelativePath().toString();
+            if(!fileCopyDetailsInternal.isDirectory() && entryName.startsWith(Flask.Constants.LIBRARIES_FOLDER)) {
+                md.reset();
+                try(InputStream is = new DigestInputStream(new FileInputStream(fileCopyDetailsInternal.getFile()), md)) {
+                    while(is.read(buffer) >= 0) {}
+                }
+                Attributes attr = manifest.getEntries().computeIfAbsent(entryName, it -> new Attributes());
+                attr.putValue(Flask.ManifestAttributes.ENTRY_HASH,
+                        String.format("%032x", new BigInteger(1, md.digest())));
+            }
+            fileCopyDetailsInternals.add(fileCopyDetailsInternal);
+        }
+    }
+
+    @Override
+    protected CopyAction createCopyAction() {
+        File destination = getArchiveFile().get().getAsFile();
+        return new CopyAction() {
+            @Override
+            @SneakyThrows
+            public WorkResult execute(CopyActionProcessingStream copyActionProcessingStream) {
+                try (ZipOutputStream zipOutputStream = new ZipOutputStream(new FileOutputStream(destination))) {
+                    Manifest manifest = new Manifest();
+                    manifest.getMainAttributes().put(Attributes.Name.MANIFEST_VERSION, "1.0");
```

Comment:
Doesn't `Manifest` contain a `MANIFEST_VERSION` attribute automatically, by default?

Reply:
I'm pretty sure it doesn't, not when you create it using the no-argument constructor (which is why I added this line)
I'll double check though

Reply:
I confirm it doesn't, you can validate with
```
#!/usr/bin/env groovy
import java.util.jar.Manifest

Manifest manifest = new Manifest()
manifest.write(System.out)
```

Reply:
OK, fair enough :+1:.